src/Razor/test/Microsoft.AspNetCore.Razor.LanguageServer.Test/Formatting/RazorFormattingTest.cs

```diff
@@ -191,7 +193,130 @@ private void IncrementCount()
         currentCount++;
     }
 }
+",
+fileKind: FileKinds.Legacy);
+        }
+
+        [Fact]
+        public async Task Layout()
+        {
+            await RunFormattingTestAsync(
+input: @"
+@layout    MyLayout
+",
+expected: @"@layout MyLayout
+");
+        }
+
+        [Fact]
+        public async Task Inherits()
+        {
+            await RunFormattingTestAsync(
+input: @"
+@inherits    MyBaseClass
+",
+expected: @"@inherits MyBaseClass
+");
+        }
+
+        [Fact]
+        public async Task Inject()
+        {
+            await RunFormattingTestAsync(
+input: @"
+@inject    MyClass     myClass
+",
+expected: @"@inject MyClass myClass
+");
+        }
+
+        [Fact]
+        public async Task Inject_TrailingWhitespace()
+        {
+            await RunFormattingTestAsync(
+input: @"
+@inject    MyClass     myClass   
+",
+expected: @"@inject MyClass myClass
+");
+        }
+
+        [Fact]
+        public async Task Attribute()
+        {
+            await RunFormattingTestAsync(
+input: @"
+@attribute     [Obsolete(   ""asdf""   , error:    false)]
+",
+expected: @"@attribute [Obsolete(""asdf"", error: false)]
 ");
         }
+
+        [Fact]
+        public async Task Model()
+        {
+            await RunFormattingTestAsync(
+input: @"
+@model    MyModel
+",
+expected: @"@model MyModel
+",
+            fileKind: FileKinds.Legacy);
+        }
+
+        [Fact]
+        public async Task Page()
+        {
+            await RunFormattingTestAsync(
+input: @"
+@page    ""MyPage""
+",
+expected: @"@page ""MyPage""
+",
+            fileKind: FileKinds.Legacy);
+        }
+
+        // Regression prevention tests:
+        [Fact]
+        public async Task Using()
+        {
+            await RunFormattingTestAsync(
+input: @"
+@using   System;
+",
+expected: @"@using   System;
+");
+        }
+
+        [Fact]
+        public async Task UsingStatic()
+        {
+            await RunFormattingTestAsync(
+input: @"
+@using  static   System.Math;
+",
+expected: @"@using  static   System.Math;
+");
+        }
+
+        [Fact]
+        public async Task TagHelpers()
+        {
+            await RunFormattingTestAsync(
+input: @"
+@addTagHelper    *,    Microsoft.AspNetCore.Mvc.TagHelpers
+@removeTagHelper    *,     Microsoft.AspNetCore.Mvc.TagHelpers
+@addTagHelper    ""*,  Microsoft.AspNetCore.Mvc.TagHelpers""
+@removeTagHelper    ""*,  Microsoft.AspNetCore.Mvc.TagHelpers""
+@tagHelperPrefix    th:
+",
+expected: @"@addTagHelper    *,    Microsoft.AspNetCore.Mvc.TagHelpers
+@removeTagHelper    *,     Microsoft.AspNetCore.Mvc.TagHelpers
+@addTagHelper    ""*,  Microsoft.AspNetCore.Mvc.TagHelpers""
+@removeTagHelper    ""*,  Microsoft.AspNetCore.Mvc.TagHelpers""
+@tagHelperPrefix    th:
```

Comment:
Are all these extra spaces really expected?

Reply:
The changes in this PR shouldn't touch tag helpers at all, so these tests are here just to make sure I don't break anything accidentally. Eventually these tests should fail (and get updated) because tag helpers (and usings) get properly formatted.